app/assets/javascripts/discourse-i18n/src/index.js

```diff
@@ -0,0 +1,384 @@
+        options.locale = this.fallbackLocale;
+        translation = this.findTranslation(scope, options);
+      }
+
+      options.ignoreMissing = false;
+
+      if (!translation && this.currentLocale() !== this.defaultLocale) {
+        options.locale = this.defaultLocale;
+        translation = this.findTranslation(scope, options);
+      }
+
+      if (!translation && this.currentLocale() !== "en") {
+        options.locale = "en";
+        translation = this.findTranslation(scope, options);
+      }
+    }
+
+    try {
+      return this.interpolate(translation, options, scope);
+    } catch (error) {
+      if (error instanceof I18nMissingInterpolationArgument) {
+        throw error;
+      } else {
+        return (
+          options.translatedFallback ||
+          this.missingTranslation(scope, null, options)
+        );
+      }
+    }
+  }
+
+  toNumber(number, options) {
+    options = this.prepareOptions(options, this.lookup("number.format"), {
+      precision: 3,
+      separator: SEPARATOR,
+      delimiter: ",",
+      strip_insignificant_zeros: false,
+    });
+
+    let negative = number < 0;
+    let string = Math.abs(number).toFixed(options.precision).toString();
+    let parts = string.split(SEPARATOR);
+    let buffer = [];
+    let formattedNumber;
+
+    number = parts[0];
+
+    while (number.length > 0) {
+      let pos = Math.max(0, number.length - 3);
+      buffer.unshift(number.slice(pos, pos + 3));
+      number = number.slice(0, -3);
+    }
+
+    formattedNumber = buffer.join(options.delimiter);
+
+    if (options.precision > 0) {
+      formattedNumber += options.separator + parts[1];
+    }
+
+    if (negative) {
+      formattedNumber = "-" + formattedNumber;
+    }
+
+    if (options.strip_insignificant_zeros) {
+      let regex = {
+        separator: new RegExp(options.separator.replace(/\./, "\\.") + "$"),
+        zeros: /0+$/,
+      };
+
+      formattedNumber = formattedNumber
+        .replace(regex.zeros, "")
+        .replace(regex.separator, "");
+    }
+
+    return formattedNumber;
+  }
+
+  toHumanSize(number, options) {
+    let kb = 1024;
+    let size = number;
+    let iterations = 0;
+    let unit, precision;
+
+    while (size >= kb && iterations < 4) {
+      size = size / kb;
+      iterations += 1;
+    }
+
+    if (iterations === 0) {
+      unit = this.t("number.human.storage_units.units.byte", { count: size });
+      precision = 0;
+    } else {
+      unit = this.t(
+        "number.human.storage_units.units." +
+          [null, "kb", "mb", "gb", "tb"][iterations]
+      );
+      precision = size - Math.floor(size) === 0 ? 0 : 1;
+    }
+
+    options = this.prepareOptions(options, {
+      precision,
+      format: this.t("number.human.storage_units.format"),
+      delimiter: "",
+    });
+
+    number = this.toNumber(size, options);
+    number = options.format.replace("%u", unit).replace("%n", number);
+
+    return number;
+  }
+
```

Comment:
My guess would be that things past this point are meant to be private, but someone else can refactor that later